planner/core/logical_plan_builder.go

```diff
@@ -3482,6 +3482,11 @@ func (b *PlanBuilder) buildSelect(ctx context.Context, sel *ast.SelectStmt) (p L
 	}
 
 	if sel.With != nil {
+		// By default, we will inline CTEs (non-recursive) into the outer query if they are called just once.
+		if !sel.With.IsRecursive {
+			inlineCTEs(sel.With.CTEs, sel)
```

Comment:
When building `with` clause, you should not change other parts. Please refer to the implementation of `view`.